packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,460 @@
+channel would be STDIN, the output channel would be STDOUT, and the channels would be additional
+pipes out.
+
+In addition, a Job has a logging channel that can be used to log messages to the original caller.
+These events need to be listened to (or forwarded) by the code scheduling the job. You can think 
+of those events as STDERR.
+
+## LifeCycle
+A Job goes through multiple LifeCycle events before its completion;
+1. `JobState.Queued`. The job was queued and is waiting to start.
+1. `JobState.Created`. The job's dependencies (see
+    ["Synchronizing and Dependencies"](#Dependencies)) are done running, and the job's argument
+    will be validated before the Job's code will be executed.
+1. `JobState.Started`. The job has been started. The job's code is then called.
+1. `JobState.Ended`. The job has ended and is done running.
+
+Each state (except Queued) corresponds to a JobEvent on the `rawOutput` observable that 
+triggers the state change. The `Queued` state is set as the job is scheduled, so there is no need
+to listen to an event. A user can listen to these events, or use the `state` member which will
+correspond to one of those. The Job implementation itself should not emit those events (they 
+will be filtered out); the Scheduler should take care of it.
+
+## `Job<OutputType>` Object
+The `Job` object is returned when scheduling a job. It contains multiple members and utilities to 
+ease the usage of jobs.
+
+1. `id`. A unique symbol that can be used as a Map key.
+1. `description`. The description of the job from the scheduler. See `JobDescription` object.
+1. `argument`. The argument value that was used to start the job.
+1. `input`. An `Observer` that can be used to send validated inputs to the Job itself.
+1. `output`. An `Observable<OutputType>` that filters out events to get only the returned output
+    of a job.
+1. `promise`. A promise that waits for the last output of a job. Returns the last value outputted
+    (or no value if there's no last value).
+1. `state`. The current state of the job (see `LifeCycle`).
+1. `progress`. A number between 0 and 1 that can be reported by the job as the progress of the job.
+1. `channels`. A map of side channels the user can listen to as `Observable`.
+1. `ping()`. A function that can be used to ping the job, receiving a `Promise` for when the ping
+    is answered.
+1. `stop()`. Sends a `stop` input to the job, which suggests to stop the job. The job itself can
+    choose to ignore this event.
+1. `rawInput`. The raw input `Observer<JobInput>`. This can be used to send messages to the
+    `context.rawInput` observable in the job. These are `JobInput` events. See
+    ["Communicating With Jobs"](#Communicating).
+1. `rawOutput`. The raw output `Observable<JobOutput>`. This can be used to listen to events from
+   the job. See ["Communicating With Jobs"](#Communicating).
+
+# Creating Jobs
+
+A job is at its core a function with a Schema associated to it;
+
```

Comment:
A job associates a function with a schema. <<*why? how is the schema used?* ...a schema that controls?/determines?/configures? xxx...>>

Reply:
Reworded.

Reply:
Yes. Thanks! (except it's just JSON Schemas, not actual values so I reworded that).

Reply:
How about: 
...JSON schemas for the input and output values, which are used to validate the types of the argument passed in.